kafka-connect-s3/src/main/java/io/confluent/connect/s3/continuum/S3Continuum.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.connect.s3.continuum;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.config.AbstractConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+import java.util.Properties;
+
+import static io.confluent.kafka.serializers.AbstractKafkaSchemaSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG;
+import static org.apache.kafka.clients.CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG;
+import static org.apache.kafka.clients.producer.ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG;
+import static org.apache.kafka.clients.producer.ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG;
+
+public class S3Continuum {
+  private static final Logger log = LoggerFactory.getLogger(S3Continuum.class);
+
+  private Producer<Object, Object> producer;
+  private String topic;
+  private int partition;
+  private ObjectMapper mapper;
+  private Schema valueSchema;
+
+  public S3Continuum(AbstractConfig config) {
+    final S3ContinuumConfigValues continuumConfig = S3ContinuumConfig.parseConfigValues(config);
+
+    if (continuumConfig.isConfigured()) {
+      Properties producerProperties = new Properties();
+      producerProperties.put(BOOTSTRAP_SERVERS_CONFIG, continuumConfig.bootstrapServers);
+      producerProperties.put(SCHEMA_REGISTRY_URL_CONFIG, continuumConfig.schemaRegistryURL);
+      producerProperties.put(KEY_SERIALIZER_CLASS_CONFIG,
+              org.apache.kafka.common.serialization.StringSerializer.class);
+      producerProperties.put(VALUE_SERIALIZER_CLASS_CONFIG,
+              continuumConfig.valueConverter);
+      this.producer = new KafkaProducer<>(producerProperties);
+
+      if (continuumConfig.schemaRegistryURL != "") {
```

Comment:
If we're relying on specifying the schema registry or not as the switch for avro vs. json, what do you think about removing value converter from the config and calling this one something like `continuum.avro.registry.url` to make it clear that it's only for avro? Or maybe we should check for the value converter class having avro in it instead of the schema registry URL?

(Technically, there's an `io.confluent.kafka.serializers.KafkaJsonSerializer` that also supports schema registry)

Reply:
Yeah, that sounds good to me.